p2p/libp2p/metrics/factory/connectionWatcherFactory.go

```diff
@@ -8,15 +8,18 @@ import (
 	"github.com/ElrondNetwork/elrond-go/p2p/libp2p/metrics"
 )
 
-const printConnectionsWatcher = "print"
-const disabledConnectionsWatcher = "disabled"
+const (
+	printConnectionsWatcher    = "print"
+	disabledConnectionsWatcher = "disabled"
+	emptyConnectionsWatcher    = ""
+)
 
 // NewConnectionsWatcher creates a new ConnectionWatcher instance based on the input parameters
 func NewConnectionsWatcher(connectionsWatcherType string, timeToLive time.Duration) (p2p.ConnectionsWatcher, error) {
 	switch connectionsWatcherType {
 	case printConnectionsWatcher:
 		return metrics.NewPrintConnectionsWatcher(timeToLive)
-	case disabledConnectionsWatcher:
+	case disabledConnectionsWatcher, emptyConnectionsWatcher:
```

Comment:
might use a new case for `emptyConnectionsWatcher` because it can only be empty due to bad configuration. therefore, we'd want to see a log warn

Reply:
I agree to be left as it is now: the config misconfiguration will be hit because the prefs.toml is not overwritten on upgrade.

Reply:
yes, the idea to add the emptyConnectionsWatcher without any error is that the prefs.toml is not overwritten and has to be added manually